codegen/gateway.go

```diff
@@ -101,32 +101,46 @@ func NewClientSpec(jsonFile string, h *PackageHelper) (*ClientSpec, error) {
 		)
 	}
 
-	clientConfigGeneric := map[string]interface{}{}
-	err = json.Unmarshal(bytes, &clientConfigGeneric)
+	classConfig := map[string]interface{}{}
+	err = json.Unmarshal(bytes, &classConfig)
 	if err != nil {
 		return nil, errors.Wrapf(
 			err, "Could not parse json file %s: ", jsonFile,
 		)
 	}
 
-	clientType := clientConfigGeneric["clientType"].(string)
-	if clientType == "http" {
+	className := classConfig["name"].(string)
+	classType := classConfig["type"].(string)
+
+	clientConfig := classConfig["config"].(map[string]interface{})
+
+	if classType == "http" {
 		// A better solution for client configs is needed here.
 		// Assuming a string map is too restrictive
-		clientConfigObj := map[string]string{}
-		err := json.Unmarshal(bytes, &clientConfigObj)
+		httpClientConfig := map[string]string{}
 
+		serializedConfig, err := json.Marshal(clientConfig)
```

Comment:
The JSON library doesn't offer an intermediate object with an Unmarshal interface. It would be great if it did. What I am trying to do is a downcast from the outer config type, and without writing the manual reflection code inline here, it's not really possible without the pre-serialize. We could obviously write a function that did the reflection generically, but I don't see much point when this isn't exactly hot path code.

The best thing I can do is supply the raw json bytes for the config and then try re-marshaling the entire config into a specific type.